src/components/Calculator.js

```diff
@@ -1,35 +1,23 @@
-import React, { Component } from 'react';
+import React from 'react';
 import NumsOperations from './NumsOperations';
-import calculate from '../logic/calculate';
 import '../logic/operate';
 
-export default class Calculator extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: 0,
-    };
+const Calculator = (props) => {
+  const { calcState, handleClick } = props;
+  const { total, next } = calcState;
+  let value = 0;
+  if (next) {
+    value = next;
+  } else if (total) {
+    value = total;
   }
 
-  clickHandler = (e) => {
-    this.setState((prevState) => calculate(prevState, e.target.value));
-  }
-
-  render() {
-    let value = 0;
-    const { total, next } = this.state;
-    if (next) {
-      value = next;
-    } else if (total) {
-      value = total;
-    }
+  return (
+    <>
+      <input className="calc-screen" placeholder="0" value={value} readOnly />
+      <NumsOperations handleClick={handleClick} />
```

Comment:
- [ ] Let's change NumsOperations to work as a stateless functional component as the requirement of this project is to refactor all components.